src/main/asciidoc/reference/elasticsearch-misc.adoc

```diff
@@ -23,50 +23,49 @@ Page<SampleEntity> sampleEntities =
 ----
 ====
 
-[[elasticsearch.scan.and.scroll]]
-== Using Scan And Scroll For Big Result Set
+[[elasticsearch.scroll]]
+== Using Scroll For Big Result Set
 
-Elasticsearch has scan and scroll feature for getting big result set in chunks. `ElasticsearchTemplate` has scan and scroll methods that can be used as below.
+Elasticsearch has a scroll API for getting big result set in chunks. `ElasticsearchTemplate` has startScroll and continueScroll methods that can be used as below.
 
-.Using Scan and Scroll
+.Using startScroll and continueScroll
 ====
 [source,java]
 ----
-SearchQuery searchQuery = new NativeSearchQueryBuilder()
-    .withQuery(matchAllQuery())
-    .withIndices("test-index")
-    .withTypes("test-type")
-    .withPageable(new PageRequest(0,1))
-    .build();
-String scrollId = elasticsearchTemplate.scan(searchQuery,1000,false);
-List<SampleEntity> sampleEntities = new ArrayList<SampleEntity>();
-boolean hasRecords = true;
-while (hasRecords){
-    Page<SampleEntity> page = elasticsearchTemplate.scroll(scrollId, 5000L , new ResultsMapper<SampleEntity>()
-    {
-        @Override
-        public Page<SampleEntity> mapResults(SearchResponse response) {
-            List<SampleEntity> chunk = new ArrayList<SampleEntity>();
-            for(SearchHit searchHit : response.getHits()){
-                if(response.getHits().getHits().length <= 0) {
-                    return null;
-                }
-                SampleEntity user = new SampleEntity();
-                user.setId(searchHit.getId());
-                user.setMessage((String)searchHit.getSource().get("message"));
-                chunk.add(user);
-            }
-            return new PageImpl<SampleEntity>(chunk);
-        }
-    });
-    if(page != null) {
-        sampleEntities.addAll(page.getContent());
-        hasRecords = page.hasNextPage();
-    }
-    else{
-        hasRecords = false;
-    }
-    }
-}            
+SearchResultMapper searchResultMapper = new SearchResultMapper() {
```

Comment:
It is not necessary to define a custom result mapper so please remove it from this basic example. There is no need to make it more complicated than it is.

Reply:
Yes I removed it , I agree that there is no point to further complicate the example.